packages/dashboards/src/lib/services/url-interaction.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from "@angular/core";
+
+@Injectable({
+    providedIn: "root",
+})
+export class UrlInteractionService {
+    public template(url: string, data: any): string{
+        const regex = new RegExp(/\$\{([a-zA-Z0-9.]*)\}/g)
+        return url.replace(regex, (match, captured) => this.evaluate(captured.split("."), data));
+    }
+
+    private evaluate(expresion: string[], data: any): string {
+        let result = data;
```

Comment:
we could even use `return expression.reduce((result, item) => result[item], data)` but I guess it might be less readable for someone

Reply:
Implemented :)